app/src/main/java/ch/epfl/sdp/blindly/main_screen/fragments/MyMatchesFragment.kt

```diff
@@ -0,0 +1,127 @@
+package ch.epfl.sdp.blindly.main_screen.fragments
+
+import android.os.Build
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.RequiresApi
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import ch.epfl.sdp.blindly.R
+import ch.epfl.sdp.blindly.database.UserRepository
+import ch.epfl.sdp.blindly.main_screen.match.my_matches.MyMatch
+import ch.epfl.sdp.blindly.main_screen.match.my_matches.MyMatchesAdapter
+import ch.epfl.sdp.blindly.user.UserHelper
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+private const val TAG = "MyMatches"
+
+
+@AndroidEntryPoint
+class MyMatchesFragment : Fragment(), MyMatchesAdapter.OnItemClickListener {
+
+    private lateinit var myMatchesRecyclerView: RecyclerView
+    private lateinit var adapter: MyMatchesAdapter
+    private lateinit var fragView: View
+
+
+    @Inject
+    lateinit var userHelper: UserHelper
+
+    @Inject
+    lateinit var userRepository: UserRepository
+
+
+    companion object {
+        private const val ARG_COUNT = "myMatchesArgs"
+        private var counter: Int? = null
+
+        /**
+         * Create a new instance of MyMatchesFragment
+         *
+         * @param counter the position of the fragment in the TabLayout
+         * @return a MyMatchesFragment
+         */
+        fun newInstance(counter: Int): MyMatchesFragment {
+            val fragment = MyMatchesFragment()
+            val args = Bundle()
+            args.putInt(ARG_COUNT, counter)
+            fragment.arguments = args
+            return fragment
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (arguments != null) {
+            counter = requireArguments().getInt(ARG_COUNT)
+        }
+
+    }
+
+
+    /**
+     * Setup the view and retrieve the profiles to show on the match activity
+     *
+     * @param inflater
+     * @param container
+     * @param savedInstanceState
+     * @return the fragment's view
+     */
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+    ): View {
+        // Inflate the layout for this fragment
+        fragView = inflater.inflate(R.layout.activity_my_matches, container, false)
+
+        //Needs to be done in a coroutine
+        viewLifecycleOwner.lifecycleScope.launch {
```

Comment:
Feel free to ignore this one, but I think it would make more sense to launch the coroutine in the UserViewModel for example and use it in the fragment. 
For example you'd instantiate the view model here and then in the ini{} of the UserViewModel you could call the getMyMatches()
And since the viewModel can be mocked, you might (?)  be able to increase the coverage for the class (but I'm really not sure)

Reply:
I'm not sure if it would be possible, in any case I managed to put query of the database in UserRepository so this part should not be where we lose coverage the most. I believe testing the adapter is a better option but I had no luck with multiple tries as mocking some stuff(like Context) resulted in NPEs or I couldn't use a FakeActivity.

Reply:
Yep I see what you mean.
And wouldn't it be possible to test the onClickListener of the Adapter via the MyMatchFragment (I think it's what Melissa and Kilian did for the AudioLibraryAdapter, testing the adapter in the RecordingActivity)

Reply:
Since MyMatchFragment calls the querying database when it's created I can't use it. I actually tried to use RecordingActivity to test MyMatchesAdapter but I had many problems. In the meantime, I managed to add a test with ApplicationProvider.getApplicationContext() and it seemed to work, let's see if it will help the coverage.

Reply:
If it's the call to userRepository.getMyMatches0 that poses a problem that's where a viewModel can help, since the latter can be mocked and the fragment could then be instantiated

Reply:
Yes it might help I believe, though I had some problems before because the matches_list I expected from the query was not returning at the correct time/scope. Maybe we can look at this further together because I'm not sure how I will apply to this usage.